plugins/poll/assets/javascripts/discourse/components/poll-voters-irv.gjs

```diff
@@ -0,0 +1,49 @@
+import Component from "@glimmer/component";
+import { eq } from "truth-helpers";
+import avatar from "discourse/helpers/bound-avatar-template";
+import icon from "discourse-common/helpers/d-icon";
+
+export default class PollVotersComponent extends Component {
+  groupVotersByRank = (voters) => {
+    return voters.reduce((groups, voter) => {
+      const rank = voter.rank;
+      if (!groups[rank]) {
+        groups[rank] = [];
+      }
```

Comment:
groups[rank] ??= [];

Reply:
This is to instantiate an empty array for each rank before we attempt to push a voter to it.

Reply:
Yes this is what my code does too

Reply:
Apologies, I misinterpreted! haha.  I will use your suggestion.

Reply:
Improved with https://github.com/discourse/discourse/pull/27155/commits/9c57524519fb33bdb422a39938b2f587b64ff938
🎓